### PR TITLE
Return errors in Publisher's error channel for ConfigurationAsyncClient

### PR DESCRIPTION
**Pull request by srnagar**
The intent of this change is to use Publisher's error channel to propagate exceptions instead of throwing Exceptions from methods that return Mono/Flux type in `ConfigurationAsyncClient` .

Other async clients will be addressed in follow-up PRs.

### CODE REVIEW

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java [hunk 1]
```diff
@@ -112,8 +113,12 @@
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<ConfigurationSetting> addSetting(ConfigurationSetting setting) {
-        return withContext(context -> addSetting(setting, context))
-            .flatMap(response -> Mono.justOrEmpty(response.getValue()));
+        try {
```
Comment (conniey) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java]: Where is the the RuntimeException being thrown from? I'm surprised that the `withContext(…)` or underlying RestProxy isn't bubbling this to downstream.

Comment (srnagar) [reply sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java]: The exception is thrown in one of the methods in this class `validateSetting(setting)`. Besides, the purpose of this change is to catch all runtime exceptions before the user code - sdk api handshake and return them in the error channel.

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java [hunk 2]
```diff
@@ -526,8 +591,12 @@
      */
     @ServiceMethod(returns = ReturnType.COLLECTION)
     public PagedFlux<ConfigurationSetting> listSettings(SettingSelector selector) {
-        return new PagedFlux<>(() -> withContext(context -> listFirstPageSettings(selector, context)),
-            continuationToken -> withContext(context -> listNextPageSettings(context, continuationToken)));
+        try {
+            return new PagedFlux<>(() -> withContext(context -> listFirstPageSettings(selector, context)),
+                continuationToken -> withContext(context -> listNextPageSettings(context, continuationToken)));
+        } catch (RuntimeException ex) {
+            return new PagedFlux<>(() -> Mono.error(logger.logExceptionAsError(Exceptions.propagate(ex))));
```
Comment (JonathanGiles) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java]: What about a `monoError` here too?

Comment (srnagar) [reply sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java]: Updated these to `monoError` as well.

**Pull request merged**